src/Symfony/Component/HttpKernel/Controller/ControllerResolver.php

```diff
@@ -167,4 +167,61 @@ protected function instantiateController($class)
     {
         return new $class();
     }
+
+    private function getControllerError($callable)
+    {
+        if (is_string($callable)) {
+            if (!function_exists($callable)) {
```

Comment:
this is broken for ``Foo::bar`` strings, as such functions won't exist and so will go there. the check for ``::`` needs to be done first